Eatery/Controllers/Eateries/Campus/ScrollableViewController.swift

```diff
@@ -0,0 +1,374 @@
+            make.top.equalToSuperview()
+            make.leading.trailing.equalToSuperview()
+        }
+
+        menuItemsView.snp.makeConstraints { make in
+            make.top.equalTo(tabBar.snp.bottom)
+            make.leading.trailing.bottom.equalToSuperview()
+        }
+
+        menuItemsStack.snp.makeConstraints { make in
+            make.top.bottom.equalToSuperview()
+            make.leading.trailing.equalToSuperview().inset(5)
+        }
+
+    }
+
+    private func createHighToLowStack() {
+        let reversedItems = Array(allItems.reversed())
+        let sortedVC = CampusEateryExpandedMenuViewController(eatery: eatery, category: "", menu: reversedItems)
+
+        menuItemsHighToLowStack = UIStackView(arrangedSubviews: [sortedVC.view])
+        menuItemsHighToLowStack.axis = .vertical
+        menuItemsView.addSubview(menuItemsHighToLowStack)
+        menuItemsHighToLowStack.isHidden = true
+
+        self.menuItemsHighToLowStack.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+    }
+
+    private func createLowToHighStack() {
+        let sortedVC = CampusEateryExpandedMenuViewController(eatery: eatery, category: "", menu: allItems)
+
+        menuItemsLowToHighStack = UIStackView(arrangedSubviews: [sortedVC.view])
+        menuItemsLowToHighStack.axis = .vertical
+        menuItemsView.addSubview(menuItemsLowToHighStack)
+        menuItemsLowToHighStack.isHidden = true
+
+        self.menuItemsLowToHighStack.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+    }
+
+    @objc func manualScroll() {
+        let currentPos = self.scrollView?.contentOffset.y
+        let viewControllerPosOnParent = self.view.frame.origin.y
+        let viewPos = originalViews[tabBar.selectedSegmentIndex].frame.minY
+        let minOffset: CGFloat = 4
+        let targetPos = viewPos - self.scrollOffset + viewControllerPosOnParent + minOffset
+        let topOfSelf = viewControllerPosOnParent - self.scrollOffset
+        let pixPS: CGFloat = 2500
+
+        self.isManualScrolling = true
+
+        if self.scrollView?.contentOffset.y ?? 0 > topOfSelf {
+            let duration = animationDuration(from: currentPos ?? 0, to: targetPos, at: pixPS)
+            UIView.animate(withDuration: duration, delay: 0, options: .curveEaseInOut) {
+                self.scrollView?.contentOffset.y = targetPos
+            }
+        } else {
+            let duration = animationDuration(from: currentPos ?? 0, to: targetPos, at: pixPS)
+            let firstProp = fabs((currentPos ?? 0) - topOfSelf) / fabs((currentPos ?? 0) - targetPos)
+            let secondProp = fabs(topOfSelf - targetPos) / fabs((currentPos ?? 0) - targetPos)
+            let firstDuration = duration * Double(firstProp)
+            let secondDuration = duration * Double(secondProp)
+
+            UIView.animate(withDuration: firstDuration, delay: 0, options: .curveEaseIn, animations: {
```

Comment:
I think some comments explaining what happens in the if and else block would be helpful given there is a lot of math and animations